api/src/exam-environment/routes/exam-environment.test.ts

```diff
@@ -0,0 +1,601 @@
+            questionSets: [],
+            needsRetake: false
+          }
+        });
+
+        const body: Static<typeof examEnvironmentPostExamAttempt.body> = {
+          attempt: mock.examAttemptSansSubmissionTimeInMS
+        };
+
+        const res = await superPost('/exam-environment/exam/attempt')
+          .set(
+            'exam-environment-authorization-token',
+            examEnvironmentAuthorizationToken
+          )
+          .send(body);
+
+        expect(res.status).toBe(200);
+
+        // Database should update attempt
+        const updatedAttempt =
+          await fastifyTestInstance.prisma.envExamAttempt.findUnique({
+            where: { id: attempt.id }
+          });
+
+        expect(updatedAttempt).toMatchObject(body.attempt);
+      });
+    });
+
+    describe('POST /exam-environment/generated-exam', () => {
+      afterEach(async () => {
+        await fastifyTestInstance.prisma.envExamAttempt.deleteMany();
+        await mock.seedEnvExam();
+      });
+
+      it('should return an error if the given exam id is invalid', async () => {
+        const body: Static<typeof examEnvironmentPostExamGeneratedExam.body> = {
+          examId: mock.oid()
+        };
+        const res = await superPost('/exam-environment/exam/generated-exam')
+          .send(body)
+          .set(
+            'exam-environment-authorization-token',
+            examEnvironmentAuthorizationToken
+          );
+
+        expect(res).toMatchObject({
+          status: 404,
+          body: {
+            code: 'FCC_ENOENT_EXAM_ENVIRONMENT_MISSING_EXAM'
+          }
+        });
+      });
+
+      xit('should return an error if the exam prerequisites are not met', async () => {
+        // TODO: Waiting on prerequisites
+      });
+
+      it('should return an error if the exam has been attempted in the last 24 hours', async () => {
+        const recentExamAttempt = {
+          ...mock.examAttempt,
+          // Set start time such that exam has expired, but 0 hours have passed
+          startTimeInMS: Date.now() - mock.exam.config.totalTimeInMS
+        };
+        await fastifyTestInstance.prisma.envExamAttempt.create({
+          data: recentExamAttempt
+        });
+
+        const body: Static<typeof examEnvironmentPostExamGeneratedExam.body> = {
+          examId: mock.examId
+        };
+
+        const res = await superPost('/exam-environment/exam/generated-exam')
+          .send(body)
+          .set(
+            'exam-environment-authorization-token',
+            examEnvironmentAuthorizationToken
+          );
+
+        expect(res).toMatchObject({
+          status: 429,
+          body: {
+            code: 'FCC_EINVAL_EXAM_ENVIRONMENT_PREREQUISITES'
+          }
+        });
+
+        await fastifyTestInstance.prisma.envExamAttempt.update({
+          where: {
+            id: recentExamAttempt.id
+          },
+          data: {
+            // Set start time such that exam has expired, but 24 hours - 1s has passed
```

Comment:
Same principle? It's almost one day since it expired?

Reply:
The idea with this test is that if we change the logic to have a cooldown of 22 hours or 25 hours, this test should fail. We might do this on purpose, but the test is ensuring the hard-coded idea of a cooldown has its bounds fully defined, and changing is not something that should _just_ happen.